internal/resources/sourcesecret/resource_sourcesecret.go

```diff
@@ -6,27 +6,399 @@ SPDX-License-Identifier: MPL-2.0
+				if clienterrors.IsNotFoundError(err) {
+					d.SetId("")
+					return "", nil, nil, err
+				}
+
+				return "", nil, nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster source secret entry, name : %s", scopedFullnameData.FullnameCluster.Name)
+			}
+
+			scopedFullnameData = &scope.ScopedFullname{
+				Scope:           commonscope.ClusterScope,
+				FullnameCluster: resp.SourceSecret.FullName,
+			}
+
+			fullName, name := scope.FlattenScope(scopedFullnameData)
+
+			if err := d.Set(nameKey, name); err != nil {
+				return "", nil, nil, err
+			}
+
+			if err := d.Set(commonscope.ScopeKey, fullName); err != nil {
+				return "", nil, nil, err
+			}
+
+			UID = resp.SourceSecret.Meta.UID
+			meta = resp.SourceSecret.Meta
+			spec = resp.SourceSecret.Spec
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			resp, err := config.TMCConnection.ClusterGroupSourcesecretResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdSourcesecretResourceServiceGet(scopedFullnameData.FullnameClusterGroup)
+			if err != nil {
+				if clienterrors.IsNotFoundError(err) {
+					d.SetId("")
+					return "", nil, nil, err
+				}
+
+				return "", nil, nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster group source secret entry, name : %s", scopedFullnameData.FullnameClusterGroup.Name)
+			}
+
+			scopedFullnameData = &scope.ScopedFullname{
+				Scope:                commonscope.ClusterGroupScope,
+				FullnameClusterGroup: resp.SourceSecret.FullName,
+			}
+
+			fullName, name := scope.FlattenScope(scopedFullnameData)
+
+			if err := d.Set(nameKey, name); err != nil {
+				return "", nil, nil, err
+			}
+
+			if err := d.Set(commonscope.ScopeKey, fullName); err != nil {
+				return "", nil, nil, err
+			}
+
+			UID = resp.SourceSecret.Meta.UID
+			meta = resp.SourceSecret.Meta
+			spec = resp.SourceSecret.Spec.AtomicSpec
+		}
+	case commonscope.UnknownScope:
+		return "", nil, nil, errors.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.CredentialTypesAllowed[:], `, `))
+	}
+
+	return UID, meta, spec, nil
+}
+
+func resourceSourcesecretCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	sourcesecretName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read source secret name")
+	}
+
+	scopedFullnameData, _ := scope.ConstructScope(d, sourcesecretName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control source secret entry; Scope full name is empty")
+	}
+
+	var (
+		UID  string
+		meta = common.ConstructMeta(d)
+	)
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, meta)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control source secret entry, name : %s", sourcesecretName))
```

Comment:
Please check the error message

Reply:
this message is fine, as it will be wrap with not enable message from server